Fix:

libhttp/url.c:258:26: error: implicitly declaring library function 'strdup' with type 'char *(const char *)' [-Werror,-Wimplicit-function-declaration]
            check(name = strdup(name));
                         ^
libhttp/url.c:258:26: note: include the header <string.h> or explicitly provide a declaration for 'strdup'
libhttp/url.c:437:7: error: implicitly declaring library function 'snprintf' with type 'int (char *, unsigned long, const char *, ...)' [-Werror,-Wimplicit-function-declaration]
      snprintf(strrchr(url->url, '\000'), 25, ":%d", url->port);
      ^
libhttp/url.c:437:7: note: include the header <stdio.h> or explicitly provide a declaration for 'snprintf'

https://github.com/shellinabox/shellinabox/issues/518
https://github.com/shellinabox/shellinabox/pull/524
--- libhttp/url.c.orig	2019-01-27 22:18:48.000000000 -0600
+++ libhttp/url.c	2024-05-05 11:32:31.000000000 -0500
@@ -45,7 +45,7 @@
 
 #include "config.h"
 
-#define _XOPEN_SOURCE 500
+#define _XOPEN_SOURCE 600
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
